### PR TITLE
password validation in KeyManager

### PR DESCRIPTION
**Pull request by debris**
please, review carefully

### CODE REVIEW

File: libethcore/KeyManager.cpp [hunk 1]
```diff
@@ -34,6 +34,76 @@ using namespace eth;
...
+{
+std::string errorDescription(PasswordValidation _validation)
+{
+	switch (_validation)
+	{
+		case PasswordValidation::Valid: return "";
+		case PasswordValidation::PasswordIsEmpty: return "Password is empty";
+		case PasswordValidation::PasswordIsToShort: return "Password is to short";
+		case PasswordValidation::PasswordsNotEqual: return "Passwords are not equal";
+	}
```
Comment (arkpar) [inline libethcore/KeyManager.cpp:50]: gcc build fails here, no return value for default case

Comment (debris) [inline libethcore/KeyManager.cpp:50]: added `default` case

File: libethcore/KeyManager.cpp [hunk 2]
```diff
@@ -352,19 +417,26 @@ void KeyManager::changeName(Address const& _address, std::string const& _name)
...
-		if (info.passwordHint.size())
-			return info.passwordHint;
-		return m_passwordHint.at(info.passHash);
-	}
-	catch (...)
+	auto it = m_keyInfo.find(_address);
+	if (it != m_keyInfo.end())
 	{
-		return EmptyString;
+		if (it->second.passwordHint.size())
```
Comment (chfast) [inline libethcore/KeyManager.cpp]: `.size()` -> `.empty()`

**Comment by chfast** [comment]
In general, it is overdesigned. Unless we need to select different validation rules in different cases _already_ please make it simple.

File: libethcore/KeyManager.h [hunk 3]
```diff
@@ -33,6 +33,37 @@ namespace eth
 {
 class PasswordUnknown: public Exception {};
 
+enum class PasswordValidation
+{
+	Valid,
+	PasswordIsEmpty,
```
Comment (chfast) [inline libethcore/KeyManager.h]: You don't have to repeat "Password" everywhere. It is enough to have {Valid, Invalid, Empty, **Too**Short}.

File: libethcore/KeyManager.h [hunk 4]
```diff
@@ -33,6 +33,37 @@ namespace eth
...
+	PasswordIsToShort,
```
Comment (subtly) [inline libethcore/KeyManager.h]: typo: Too

File: libethcore/KeyManager.cpp [hunk 5]
```diff
@@ -34,6 +34,77 @@ using namespace eth;
...
+{
+namespace eth
+{
+std::string errorDescription(PasswordValidation _validation)
+{
+	switch (_validation)
+	{
+		case PasswordValidation::Valid: return "";
+		case PasswordValidation::PasswordIsEmpty: return "Password is empty";
+		case PasswordValidation::PasswordIsToShort: return "Password is to short";
```
Comment (subtly) [inline libethcore/KeyManager.cpp]: typo: Too

**Comment by chfast** [comment]
@debris, any your response to the comments?

**Comment by debris** [comment]
Thanks @chfast for review! I simplified pr and applied your suggestions.

File: libethcore/KeyManager.h [hunk 6]
```diff
@@ -33,6 +33,22 @@ namespace eth
...
+{
+	Valid,
+	Empty,
+	TooShort,
+	NotEqual
+};
+
+std::string errorDescription(PasswordValidation _validation);
+
+class PasswordValidator
```
Comment (gavofyork) [inline libethcore/KeyManager.h]: why a memberless class and not just a function, or at the least a static method?

File: libethcore/KeyManager.h [hunk 7]
```diff
@@ -33,6 +33,22 @@ namespace eth
...
+{
+public:
+	PasswordValidation check(std::string const& _s1, std::string const& _s2) const;
```
Comment (gavofyork) [inline libethcore/KeyManager.h]: what uses this? why does it belong in KeyManager.cpp?

Comment (debris) [inline libethcore/KeyManager.h]: it's used in https://github.com/ethereum/alethzero/pull/71 . I wanted all executables to eventually use the same password validation

File: libethcore/KeyManager.h (hunk 6 above)
Comment (debris) [inline libethcore/KeyManager.h]: I altered it. Not it's just a plain function

**Comment by chriseth** [comment]
Closing as unmaintained and unused.

**Pull request closed without merging**